tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java

```diff
@@ -343,9 +509,70 @@ private Credential createCredential(final String secret, String entryType) {
         }
     }
 
+    private KeyPairCredential createKeyPairCredential(String algorithm, int size, String sshDirectory, String knownHosts) throws NoSuchAlgorithmException {
+        KeyPairGenerator keyPairGenerator;
+
+        switch (algorithm) {
+            case Pem.RSA_ALGORITHM: {
+                /* Size must range from 512 to 16384. Default size: 2048
+                 * see: https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2
+                 */
+                size = (512 < size && size < 16384) ? size : 2048;
```

Comment:
Looks like the ranges should be inclusive so ```<``` should be ```<=``` here and below.

Reply:
Fixed, thanks!